src/main/java/makaroshyna/onlinebookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,74 @@
+package makaroshyna.onlinebookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import makaroshyna.onlinebookstore.dto.cartitem.CartItemResponseDto;
+import makaroshyna.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import makaroshyna.onlinebookstore.dto.cartitem.UpdateCartItemRequestDto;
+import makaroshyna.onlinebookstore.dto.shoppingcart.ShoppingCartResponseDto;
+import makaroshyna.onlinebookstore.model.User;
+import makaroshyna.onlinebookstore.service.shoppingcart.ShoppingCartService;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping carts")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
Same for other endpoints as well
```suggestion
    @PreAuthorize("hasRole('USER')")
```
if it does not work remove the `ROLE_` from the getter in Role entity